src/exceedances/job/views/exceedance.py

```diff
@@ -0,0 +1,387 @@
+    the specified start date and end date for the specified time frequency.
+
+    Args:
+        device_code (str): the code used to identify a device.
+        start_date (datetime): the datetime from which observations to be returned should start(lower boundary). 
+        end_date (datetime): the datetime from which observations to be returned should end(upper boundary).
+        frequency (str): the frequency of the observataions i.e. hourly, daily, monthly.
+        pollutant (str): the pollutant whose observatations are to be returned i.e. PM 2.5, PM 10, NO2.
+    Returns:
+        A list of the data(pollutant values & their corresponding time) for the specified pollutant from the device with the specified code observed between
+    the specified start date and end date for the specified time frequency.
+
+    """
+    if start_date == None:
+        start = convert_date.str_to_date_find('2019-06-01T00:00:00Z')
+    else:
+        start = convert_date.str_to_date(start_date)
+    if end_date == None:
+        end = datetime.now()
+    else:
+
+        end = convert_date.str_to_date(end_date)
+
+    query = {'$match': {'deviceCode': device_code,
+                        'time': {'$lte': end, '$gte': start}}}
+
+    if pollutant == 'PM 10':
+        projection = {'$project': {'_id': 0,
+                                   'time': {'$dateToString': {'format': '%Y-%m-%dT%H:%M:%S%z', 'date': '$time', 'timezone': 'Africa/Kampala'}},
+                                   'pollutant_value': {'$round': ['$characteristics.pm10ConcMass.value', 2]}}}
+    elif pollutant == 'NO2':
+        projection = {'$project': {'_id': 0,
+                                   'time': {'$dateToString': {'format': '%Y-%m-%dT%H:%M:%S%z', 'date': '$time', 'timezone': 'Africa/Kampala'}},
+                                   'pollutant_value': {'$round': ['$characteristics.no2Conc.value', 2]}}}
+    else:
+        projection = {'$project': {'_id': 0,
+                                   'time': {'$dateToString': {'format': '%Y-%m-%dT%H:%M:%S%z', 'date': '$time', 'timezone': 'Africa/Kampala'}},
+                                   'pollutant_value': {'$round': ['$characteristics.pm2_5ConcMass.value', 2]}}}
+
+    if frequency == 'hourly':
+        # shall just query from the Events collection accordingly.
+        records = db.device_hourly_measurements.aggregate([query, projection])
+    elif frequency == 'monthly':
+        results = list(
+            db.device_daily_measurements.aggregate([query, projection]))
+        # because of the way data is stored, there will be no need to resample
+        records = resample_timeseries_data(results, 'M', 'time', 2)
+    else:
+        records = db.device_daily_measurements.aggregate([query, projection])
+
+    return list(records)
+
+
+def save_device_daily_exceedences(data):
+    """
+    # this ones moves to the exceedances model/entity
+    """
+    for i in data:
+        db.device_daily_exceedences.insert_one(i)
```

Comment:
this is the name of the collection which stores the values.  @ja-odur See screenshot....
<img width="624" alt="Screenshot 2021-06-25 at 12 40 39" src="https://user-images.githubusercontent.com/1590213/123405523-95db3c00-d5b2-11eb-85e5-3413c4c3b7a3.png">

Reply:
Noted. Are gonna maintain the same structure @Baalmart